bc/core/utils/faker_legal_citations.py

```diff
@@ -0,0 +1,264 @@
+        return f"{plaint} v. {defend}"
+
+    @staticmethod
+    def case_party_name() -> str:
+        """
+        Make a name that could be a party to a legal case.
+        The name has a 50/50 chance of being either a person or a company.
+
+        :returns: the fake name
+        :rtype: str
+        """
+        do_company = random.choice([True, False])
+        if do_company:
+            return _faker.company()
+        else:
+            return _faker.last_name()
+
+    @staticmethod
+    def court_name() -> str:
+        """
+        Make a court names like
+          - First circuit for the zoo
+          - District court of albatross
+          - Appeals court of eczema
+
+        Basic format is <section> <connector> <whole>
+
+        :returns: the fake court name
+        :rtype: str
+        """
+        section = random.choice(
+            [
+                "Thirteenth circuit",
+                "Forty-Second circuit",
+                "District court",
+                "Appeals court",
+                "Superior court",
+            ]
+        )
+        connector = random.choice(["of the", "for the"])
+        whole = random.choice(
+            [
+                "Zoo",
+                "Medical Worries",
+                "Programming Horrors",
+                "dragons",
+                "Dirty Dishes",
+                "Eruptanyom",  # Kelvin's pretend world
+            ]
+        )
+        return " ".join([section, connector, whole])
+
+    def docket_number(self) -> str:
+        """
+        Make either a simple docket number or a federal district docket
+        number. There is an equal chance (50/50) of returning one or the other.
+
+        :returns: the fake docket number
+        :rtype:   str
+        """
+        use_simple = random.choice([True, False])
+        if use_simple:
+            return self.simple_docket_number()
+        else:
+            return self.federal_district_docket_number()
+
+    @staticmethod
+    def simple_docket_number() -> str:
+        """
+        Make a docket number of the form NN-NXXXX, where:
+             * 'N' is a number 1 - 9,
+             * 'X' is a number 0-9
+
+            Examples:
+                '17-78721'
+                '90-20018'
+
+        :returns: the fake  docket number
+        :rtype:   str
+        """
+        return _faker.numerify(text="%%-%####")
+
+    @staticmethod
+    def federal_district_docket_number() -> str:
+        """
+        Make a docket number like you'd see in a district court
+            of the form <office>:<year>-<2 characters>-<5 digits>
+
+        Ex: "2:13-cv-03239"
+
+        :returns: a docket number
+        :rtype: str
+        """
+        office = random.randint(1, 7)
+        year = random.randint(0, 99)
+        letters = random.choice(["cv", "bk", "cr", "ms"])
+        number = random.randint(1, 200_000)
+        return f"{office}:{year:02}-{letters}-{number:05}"
+
+    @staticmethod
+    def humanized_join(
```

Comment:
I don't see where this is being used, but I think it's buggy wrt to how it handles spaces around conjunctions. I think as it is presently, it'd create double spaces:

> `1, 2,  and  3`

I'd suggest having the code handle adding the spaces properly, and not to ask the user to remember to do ` and `. That feels pretty wonky to me.